tgui/packages/tgui/interfaces/GeneTek/tabs/ResearchTab.tsx

```diff
@@ -84,24 +84,28 @@ export const ResearchTab = (props) => {
         {availableResearch.map((ar, tier) => (
           <Section key={tier} title={`Tier ${tier + 1}`}>
             {ar.length
-              ? ar.map((r) => (
-                  <Section
-                    key={r.ref}
-                    title={researchTree[r.ref]?.name}
-                    buttons={
-                      <Button
-                        icon="flask"
-                        disabled={materialCur < r.cost}
-                        onClick={() => act('research', { ref: r.ref })}
-                        color="teal"
+              ? ar.map(
+                  (r) =>
+                    r.ref &&
+                    r.cost && (
+                      <Section
+                        key={r.ref}
```

Comment:
Sanity check on the `.map` returning a keyed component sometimes, and presumably `false` or `null` in others, when there's also a `.length` check earlier?

Would it be better to `.filter`, then use that for the `.length` check and then the `.map` as you have it?